test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+                        || element.getKind() == ElementKind.METHOD
+                        || element.getKind() == ElementKind.CONSTRUCTOR)
+                .forEach(element -> checkElement(te, element, types, javadocHelper, version, elementUtils));
+        te.getEnclosedElements().stream()
+                .filter(element -> element.getKind().isDeclaredType())
+                .map(TypeElement.class::cast)
+                .forEach(nestedClass -> analyzeClassCheck(nestedClass, currentjdkVersion, javadocHelper, types, elementUtils));
+    }
+
+
+    private void checkElement(Element explicitOwner, Element element, Types types,
+                              EffectiveSourceSinceHelper javadocHelper, String currentVersion, Elements elementUtils) {
+        String uniqueId = getElementName(explicitOwner, element, types);
+
+        if (element.getKind() == ElementKind.METHOD &&
+                element.getEnclosingElement().getKind() == ElementKind.ENUM &&
+                (uniqueId.contains(".values()") || uniqueId.contains(".valueOf(java.lang.String)"))) {
+            //mandated enum type methods
+            return;
+        }
+        String sinceVersion = javadocHelper.effectiveSinceVersion(explicitOwner, element, types, elementUtils).toString();
+        IntroducedIn mappedVersion = classDictionary.get(uniqueId);
+        String realMappedVersion = null;
+        try {
+            realMappedVersion = isPreview(element, uniqueId, currentVersion) ?
+                    mappedVersion.introducedPreview :
+                    mappedVersion.introducedStable;
+        } catch (Exception e) {
+            wrongTagsList.add("For element " + element + "mappedVersion" + mappedVersion + " is null" + e + "\n");
+        }
+        checkEquals(sinceVersion, realMappedVersion, uniqueId);
+    }
+
+    private Version extractSinceVersionFromText(String documentation) {
+        Pattern pattern = Pattern.compile("@since\\s+(\\d+(?:\\.\\d+)?)");
+        Matcher matcher = pattern.matcher(documentation);
+        if (matcher.find()) {
+            String versionString = matcher.group(1);
+            try {
+                if (versionString.equals("1.0")) {
+                    versionString = "1"; //ended up being necessary
+                } else if (versionString.startsWith("1.")) {
+                    versionString = versionString.substring(2);
+                }
+                return Version.parse(versionString);
+            } catch (NumberFormatException ex) {
+                wrongTagsList.add("`@since` value that cannot be parsed: " + versionString + "\n");
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    private void checkEquals(String sinceVersion, String mappedVersion, String id) {
+        if (sinceVersion == null || mappedVersion == null) {
+            wrongTagsList.add("For " + id + " NULL value for either mapped or real `@since` . mapped version is="
```

Comment:
Here and elsewhere, instead of `"For " + name + ....`  consider using the format `name + ":  " + ...` for the format of the messages

Reply:
Fixed it.